app/error_handler.py

```diff
@@ -0,0 +1,61 @@
+"""Error handler module.
+
+
+"""
+import logging
+
+from app.configuration import Configuration
+from app.mqtt_writer import MqttWriter
+from app.exceptions import InvalidArgumentError
+from app.message import ErrorMessage
+import paho.mqtt.client as mqtt
+
+
+class ErrorHandler:
+    class Category:
+        MODBUS_ERROR = "ModbusError"
+        MQTT_ERROR = "MQTTError"
+        INVALID_MESSAGE = "InvalidMessage"
+        UNKNOWN_COMMAND = "UnknownCommand"
+
+    @classmethod
+    def from_config(cls, config: Configuration, mqtt_client: mqtt.Client = None):
+        mqtt_settings = config.get_mqtt_settings()
+        if mqtt_settings.pub_errors:
+            if not mqtt_client:
```

Comment:
This double nested if could be avoided if we always inject the mqtt client.

My preference remains on avoiding partial state but I won't block this pr. It's mostly where I think we should be, so I'm ok proceeding with this

Reply:
BTW one reason why I have this preference is because partial state is always evaluated on runtime rather than bootstrap time. By enforcing all dependencies to be instantiated as part of the constructor, you can do static analysis that shows errors. The app doesn't start if there is an error here. However, with partial state, the app does starts and is only on runtime when you can the accidental "null pointer exceptions"

Reply:
I see the point, but do you think it's OK to always require the MQTT client in the constructor, even if the site is configured not to publish error messages? I guess it doesn't affect the way we actually init ErrorHandler, but in general it seems to me that a class shouldn't require a dependency if it knows at bootstrap time (at least after reading config) that it's not going to use it.

Reply:
What's the problem of doing that though?
The object exists regardless. It's indeed the same object after all.
It's just a pointer pointing to a real memory allocation rather than pointing nowhere.

On the other hand, the class becomes simpler. As a developer you don't need to worry to check if the dependencies exist or not. This often leads to less accidental runtime errors

Reply:
But I have approved your pr. I don't feel we must address this right now. Feel free to move on with the pr as is.

We may revisit this in the future but not a blocker now

Reply:
Yes, thanks for that. Just trying to understand the reasoning as it doesn't feel intuitive to me. I'll attach those screenshots then merge this one, but might come back to this in #39